host/tests/dsDisplay_L3_Tests/dsDisplay_test01_GetDisplayEDID.py

```diff
@@ -0,0 +1,181 @@
+from raft.framework.plugins.ut_raft.utUserResponse import utUserResponse
+
+class dsDisplay_test01_GetDisplayEDID(utHelperClass):
+
+    """
+    Test class to retrieve and verify the EDID (Extended Display Identification Data) from a display.
+
+    This class uses the `dsDisplayClass` to interact with the device's display,
+    downloading necessary test assets, retrieving the EDID, and performing verification.
+    """
+
+    # Class variables
+    testName  = "test01_GetDisplayEDID"
+    testSetupPath = dir_path + "/dsDisplay_L3_testSetup.yml"
+    moduleName = "dsDisplay"
+    rackDevice = "dut"
+
+    def __init__(self):
+        """
+        Initializes the test03_GetDisplayAspectRatio test .
+
+        Args:
+            None.
+        """
+        super().__init__(self.testName, '1')
+
+        # Test Setup configuration file
+        self.testSetup = ConfigRead(self.testSetupPath, self.moduleName)
+
+        # Open Session for hal test
+        self.hal_session = self.dut.getConsoleSession("ssh_hal_test")
+
+        # Create user response Class
+        self.testUserResponse = utUserResponse()
+
+        # Get path to device profile file
+        self.deviceProfile = dir_path + "/" + self.cpe.get("test").get("profile")
+
+    def testDownloadAssets(self):
+        """
+        Downloads the artifacts in test-setup configuration file to the dut.
+
+        Args:
+            None.
+        """
+
+        self.deviceDownloadPath = self.cpe.get("target_directory")
+
+        test = self.testSetup.get("assets").get("device").get(self.testName)
+
+        # Download test artifacts to device
+        url = test.get("artifacts")
+        if url is not None:
+            self.downloadToDevice(url, self.deviceDownloadPath, self.rackDevice)
+
+    def testRunPrerequisites(self):
+        """
+        Runs Prerequisite commands listed in test-setup configuration file on the dut.
+
+        Args:
+            None.
+        """
+
+        #Run test specific commands
+        test = self.testSetup.get("assets").get("device").get(self.testName)
+        cmds = test.get("execute")
+        if cmds is not None:
+            for cmd in cmds:
+                self.writeCommands(cmd)
+
+    #TODO: Current version supports only manual verification.
+    def testGetEDID(self, manual=False, port="dsVIDEOPORT_TYPE_HDMI" , index:int = 0):
+        """
+        Gets the EDID of the display.
+        Args:
+            manual (bool, optional): Manual verification (True: manual, False: other verification methods).
+                                    Defaults to other verification methods
+                                    port (str, optional): The type of video port to retrieve the EDID.
+                                    index (int, optional): The index of the port to retrieve the EDID.
+        Returns:
+            bool
+        """
+        # Get the EDID from the display
+        edid = self.testdsDisplay.getEdid(port, index)
+
+        if manual:
```

Comment:
verification should not be manual. list the required edid details of connected displays' in yaml and use that information to validate the edid returned by getEdid